turbinia/turbiniactl.py

```diff
@@ -753,26 +726,35 @@ def main():
     request = TurbiniaRequest(
         request_id=request_id, requester=getpass.getuser())
     request.evidence.append(evidence_)
-    if filter_patterns:
-      request.recipe['filter_patterns'] = filter_patterns
-    if args.jobs_denylist:
-      request.recipe['jobs_denylist'] = args.jobs_denylist
-    if args.jobs_allowlist:
-      request.recipe['jobs_allowlist'] = args.jobs_allowlist
-    if yara_rules:
-      request.recipe['yara_rules'] = yara_rules
-    if args.debug_tasks:
-      request.recipe['debug_tasks'] = args.debug_tasks
-    if args.recipe_config:
-      for pair in args.recipe_config:
-        try:
-          key, value = pair.split('=')
-        except ValueError as exception:
-          log.error(
-              'Could not parse key=value pair [{0:s}] from recipe config '
-              '{1!s}: {2!s}'.format(pair, args.recipe_config, exception))
-          sys.exit(1)
-        request.recipe[key] = value
+
+    if args.recipe:
+      if (args.jobs_denylist or args.jobs_allowlist or
+          args.filter_patterns_file or args.yara_rules_file):
+        raise TurbiniaException(
+            'Specifying a recipe is incompatible with defining'
+            ' jobs allow/deny lists, yara rules or a patterns file separately.')
+      try:
+        recipe_obj = TurbiniaRecipe(
+            os.path.join(config.RECIPE_FILE_DIR, args.recipe))
+      except IOError as e:
+        log.warning(
+            'Cannot open file {0:s} [{1!s}]'.format(args.yara_rules_file, e))
+        sys.exit(1)
+      recipe_obj.load()
```

Comment:
I think there is a typo there. the try block seeks to catch a potential exception thrown when the file for the recipe is not found. Otherwise the TurbiniaRecipe class uses the file helpers that in turn have exception catching blocks.